api/app/helpers/seeds_handler.rb

```diff
@@ -0,0 +1,259 @@
+module SeedsHandler
+    include FakeData
+    def request(action, path, params = nil)
+        '''
+        Returns the result of calling the API path using action and given params
+
+        action: :get, :post, :put, :delete
+        path: no need to include "/api/v1" portion of the route
+        params: body of API call
+        '''
+        begin
+            env = { method: action, params: params }
+            mock_session = Rack::MockSession.new(Rails.application)
+            session = Rack::Test::Session.new(mock_session)
+            session.request("/api/v1#{path}", env)
+            return JSON.parse(mock_session.last_response.body, symbolize_names: true)
+        rescue JSON::ParserError
+            error = "#{action.to_s.upcase} #{path} is an invalid path."
+            return {status: 'error', message: error, payload: {}}
+        end
+    end
+
+    def read_json(file, log)
+        '''
+        Returns a JSON with the file content and a log containing messages if any
+
+        file: located relative to seed_data_dir
+        log: records error messages
+        '''
+        seed_data_dir = Rails.root.join('db', 'seed')
+        file_dir = seed_data_dir+file
+        begin
+            data = File.read(file_dir)
+            return JSON.parse(data, symbolize_names: true), log
+        rescue Errno::ENOENT => e
+            log.push(e)
+            return {}, log
+        rescue JSON::ParserError 
+            log.push("#{file_dir} is not a valid JSON file.")
+            return {}, log
+        end
+    end
+
+    def insert_data(chaining, file)
+        '''
+        Inserts JSON mock data into the routes defined in chaining.
+
+        chaining: an array with JSON as elements. Each element contains
+            get: route to get all entries for a table
+            create: route to create an entry in a specific table
+            unique: an array of attributes that makes an entry unique in a table
+        file: JSON file name containg seed data to be inserted
+        '''
+        log = []
+        empty_db, log = empty_records(chaining, log)
+        if empty_db
+            if file
+                json, log = read_json(file, log)
+                valid_templates, log = check_templates(json, log)
+                if valid_templates
+                    log = post_data(chaining, json, log)
+                end
+            else
+                log.push('No JSON file was given.')
+            end
+        end
+        puts log
```

Comment:
Isn't there a more offical way to log something?

Reply:
There is. However, the problem with that it is the logging goes immediately into the log of the rest of the stuff Rack is logging at the same time, so any message that's relevant to the user kind of gets lost in between other really long messages. This is why I chose to accumulate the log messages because I print them.